### PR TITLE
refactor(NODE-5778): change kConnection to connection, split polling and streaming functions, define exhaustCommand

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

Some minor refactors to the monitoring class 

##### Is there new documentation needed for these changes?

No

#### What is the motivation for this change?

To pull them out of the "feature" work of adding streaming support to the new monitoring class.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: src/cmap/connection.ts [hunk 1]
```diff
@@ -562,6 +562,15 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
       callback(err);
     }
   }
+
+  moreToComeCommand(
```
Comment (baileympearson) [inline src/cmap/connection.ts:566]: Are we attached to the name `moreToComeCommand`?  The publicly documented many-to-one behavior is referrred to as "exhaust" (exhaust allowed, exhaust cursors, etc), and `moreToCome` is just an implementation detail of exhaust support.   `exhaustCommand` might make more sense

Comment (durran) [inline src/cmap/connection.ts:566]: Just out of curiosity, what was the need for this? Are there upcoming changes that will alter the behaviour of this method?

File: src/cmap/commands.ts [hunk 2]
```diff
@@ -317,6 +317,9 @@ export class OpQueryResponse {
   bsonRegExp?: boolean;
   index?: number;
 
+  /** moreToCome is an OP_MSG only concept */
+  moreToCome = false;
```
Comment (baileympearson) [inline src/cmap/commands.ts:321]: why was this necessary?  I'd prefer to keep our existing `!('moreToCome' in response)` check than add an OpMsg-specific concept to our QpQuery classes.

also - it shouldn't be necessary on the request, right?

File: src/cmap/connection.ts (hunk 1 above)
Comment (nbbeeken) [reply src/cmap/connection.ts]: No, this method will remain as is until we remove the old connection class entirely. This is so the monitor can be refactored to support either Connection or ModernConnection without needing to gaurd on which instance of the connection class it is using. I added the empty method to ModernConnection to make it a bit more clear, the next segment of work will fill out that method and make Monitors start testing with the new connection class.

Comment (nbbeeken) [reply src/cmap/connection.ts]: SGTM

File: src/cmap/commands.ts (hunk 2 above)
Comment (nbbeeken) [reply src/cmap/commands.ts:321]: It is not necessary, but neither is the `in` check, which is artificially introduced by typescript. Since OP msg vs query isn't something our driver code generally needs to discriminate it would be ideal if we could create an interface that both classes implement, then functions that do not care which protocol is being used can depend on that interface. 

That interface idea would be nice but a bit more scope than is appropriate for one condition improvement. I do not feel too strongly.

File: src/cmap/connection.ts (hunk 1 above)
Comment (durran) [reply src/cmap/connection.ts]: Ah ok, thanks for the explanation.

**Review by durran — APPROVED** [review]

File: src/sdam/monitor.ts [hunk 3]
```diff
@@ -94,21 +92,17 @@ export class Monitor extends TypedEventEmitter<MonitorEvents> {
   connectOptions: ConnectionOptions;
   isRunningInFaasEnv: boolean;
   [kServer]: Server;
-  [kConnection]?: Connection;
+  connection: Connection | null;
```
Comment (alenakhineika) [inline src/sdam/monitor.ts:95]: What is the motivation behind changing kConnection to connection? Or the backstory of why it was a symbol?

**Review by alenakhineika — APPROVED** [review]

File: src/sdam/monitor.ts (hunk 3 above)
Comment (baileympearson) [reply src/sdam/monitor.ts:95]: In the pre-TS days of the driver, there were issues with users accessing driver internal properties because we had no way to mark them internal.  So we switched to using symbol properties, similar to how Nodejs handles its internal properties, because although they're still accessible to users, it's less accessible.

Now that we have TS with the `private` keyword, generated TS-doc that strips out anything with an `@internal` tag and someday JS private properties, we don't really need symbol properties anymore.  

We're slowly removing them because they cause headaches for us:

- harder to access in tests
- no cmd+click to jump to definition of a class property
- doc comments don't appear on class properties that use symbols when you hover over the property
- etc

**Review by baileympearson — APPROVED** [review]

File: src/sdam/monitor.ts (hunk 3 above)
Comment (alenakhineika) [reply src/sdam/monitor.ts:95]: Got it! Thx for the context!
